draft-irtf-cfrg-hash-to-curve.md

```diff
@@ -1991,36 +2012,201 @@ conditions.
+- Z: -2
+- h\_eff: 1
+
+## Suites for curve25519 and edwards25519 {#suites-25519}
+
+This section defines ciphersuites for curve25519 and edwards25519 {{RFC7748}}.
+
+The suites curve25519-SHA256-ELL2-RO and curve25519-SHA256-ELL2-NU
+share the following parameters, in addition to the common parameters below.
+
+- E: y^2 = x^3 + 486662 * x^2 + x
+- f: Elligator 2 method, {{elligator2}}
+
+The suites edwards25519-SHA256-EDELL2-RO and edwards25519-SHA256-EDELL2-NU
+share the following parameters, in addition to the common parameters below.
+
+- E: -x^2 + y^2 = 1 + d * x^2 * y^2, where
+  - d = 0x52036cee2b6ffe738cc740797779e89800700a4d4141d8ab75eb4dca135978a3
+- f: Twisted Edwards Elligator 2 method, {{ell2edwards}}
+- M: curve25519 defined in {{RFC7748}}, Section 4.1
+- rational\_map: the birational map defined in {{RFC7748}}, Section 4.1
+
+The common parameters for all of the above suites are:
+
+- p: 2^255 - 19
+- m: 1
+- H: SHA-256
+- W: 2
+- Z: 2
+- h\_eff: 8
+
+## Suites for curve448 and edwards448 {#suites-448}
+
+This section defines ciphersuites for curve448 and edwards448 {{RFC7748}}.
+
+The suites curve448-SHA512-ELL2-RO and curve448-SHA512-ELL2-NU
+share the following parameters, in addition to the common parameters below.
+
+- E: y^2 = x^3 + 156326 * x^2 + x
+- f: Elligator 2 method, {{elligator2}}
+
+The suites edwards448-SHA512-EDELL2-RO and edwards448-SHA512-EDELL2-NU
+share the following parameters, in addition to the common parameters below.
+
+- E: x^2 + y^2 = 1 - 39081 * x^2 * y^2
+- f: Twisted Edwards Elligator 2 method, {{ell2edwards}}
+- M: curve448, defined in {{RFC7748}}, Section 4.2
+- rational\_map: the 4-isogeny map defined in {{RFC7748}}, Section 4.2
+
+The common parameters for all of the above suites are:
+
+- p: 2^448 - 2^224 - 1
+- m: 1
+- H: SHA-512
+- W: 2
+- Z: -1
+- h\_eff: 4
+
+## Suites for SECP256K1 {#suites-secp256k1}
+
+The suites SECP256K1-SHA256-SVDW-RO and SECP256K1-SHA256-SVDW-NU
+are defined for the SECP256K1 elliptic curve {{SEC2}}.
+These suites share the following parameters:
+
+- E: y^2 = x^3 + 7
+- p: 2^256 - 2^32 - 2^9 - 2^8 - 2^7 - 2^6 - 2^4 - 1
+- m: 1
+- H: SHA-256
+- W: 2
+- f: Shallue-van de Woestijne method, {{swpairing}}
+- Z: 1
+- h\_eff: 1
+
+## Suites for BLS12-381 {#suites-bls12381}
+
+This section defines ciphersuites for groups G1 and G2 of
+the BLS12-381 elliptic curve {{draft-yonezawa-pfc-01}}.
+
+The suites BLS12\_381\_G1-SHA256-SSWU-RO and BLS12\_381\_G1-SHA256-SSWU-NU
+share the following parameters, in addition to the common parameters below.
+
+- E: y^2 = x^3 + 4
+- m: 1
+- Z: -1
+- E': the 11-isogenous curve from {{WB19}}; see {{appx-bls12381-g1}}
+- iso\_map: the isogeny map to E' from {{WB19}}; see {{appx-bls12381-g1}}
+- h\_eff: 0xd201000000010001
 
+The suites BLS12\_381\_G2-SHA256-SSWU-RO and BLS12\_381\_G2-SHA256-SSWU-NU
+share the following parameters, in addition to the common parameters below.
+
+- E: y^2 = x^3 + 4 * (1 + i), where
```

Comment:
I think it is better to define the extension field and its irreducible polynomial before introduce curve parameters

Reply:
Better now?

Reply:
Yes, it is.